conan/tools/microsoft/msbuild.py

```diff
@@ -41,10 +41,15 @@ def command(self, sln):
         if maxcpucount:
             cmd += " /m:{}".format(maxcpucount)
 
+        if targets:
+            if not isinstance(targets, list):
```

Comment:
why? I would allow any `Iterable`, so generator/co-routine could be used here

Reply:
I prefer interfaces to be as defined, limited, and constrained as possible, and open them up when real use cases happen. If you have a generator just call the method ``msbuild.build(...targets=list(yourstuff))``

Example: If we allow an ``Iterable``, it happens that a string is ``Iterable``, so users passing a string ``target="mytarget"``, will get a ``/target=m;y;t;a;r;g;e;t`` that will produce a more difficult to interpret later at build output. You can say, now you have to handle the ``string`` case differently in the method, and that starts to add code to the method, that needs to be tested, documented, and maintained, and this is exactly what I want to avoid